### PR TITLE
Override babelrc for the node related code

### PR DESCRIPTION
**Pull request by igor-dv**
Issue: We may optimize the Node related code to be transpired for Node. That will reduce the package size a bit and will ease a debugging

## What I did
~~I've added the Node specific `.babelrc` in a few places~~
Overridden the root babelrc for the Node related code

### CODE REVIEW

**Comment by Hypnosphi** [comment]
You can use `overrides` instead

**Review by ndelangen — APPROVED** [review]

File: .babelrc [hunk 1]
```diff
@@ -28,6 +28,25 @@
...
+      "presets": [
+        [
+          "@babel/preset-env",
+          {
+            "targets": {
+              "node": "8.11"
+            }
+          }
+        ]
+      ]
```
[the remark this answers is not in the document]
Comment (Hypnosphi) [reply .babelrc]: Or maybe you even need to explicitly disable each particular plugin =(

[the remark this answers is not in the document]
Comment (Hypnosphi) [reply .babelrc]: Like this:

```
"presets": [
  [
    "@babel/preset-env",
    {...}
  ],
  ["@babel/preset-react", false],
  ["@babel/preset-flow", false],
]
```

and same for `plugins`

File: .babelrc [hunk 2]
```diff
@@ -28,6 +28,25 @@
...
       ]
+    },
+    {
+      "test": [
+        "./lib/core/src/server",
+        "./lib/node-logger",
+        "./addons/storyshots",
+        "./addons/storysource/src/loader",
+        "./app/**/src/server/**"
+      ],
```
[the remark this answers is not in the document]
Comment (igor-dv) [reply .babelrc]: Added codemod. 
`lib/cli` seems not transpiled like other packages (there is no src dir in)

[the remark this answers is not in the document]
Comment (Hypnosphi) [reply .babelrc]: it's using `babel-register`

File: .babelrc (hunk 1 above)
[the remark this answers is not in the document]
Comment (igor-dv) [reply .babelrc]: I think we still need "@babel/plugin-proposal-class-properties" and "@babel/plugin-proposal-export-default-from" for node.

The question is do I need to remove react and flow from the places where react and flow will probably not used?

File: .babelrc (hunk 2 above)
[the remark this answers is not in the document]
Comment (igor-dv) [reply .babelrc]: Yeah, but the "scripts/compile-js" is not handling this I think... I mean, do you think I need to change this?

File: .babelrc (hunk 1 above)
[the remark this answers is not in the document]
Comment (Hypnosphi) [reply .babelrc]: Maybe not =) There's a chance that just changing `env` settings is enough

File: .babelrc (hunk 2 above)
[the remark this answers is not in the document]
Comment (Hypnosphi) [reply .babelrc]: Less configs is better, I think

[the remark this answers is not in the document]
Comment (igor-dv) [reply .babelrc]: Cool. Will wait for #4236 to be finished. Then will move it to `src` dir.

**Pull request merged**